src/main/java/org/jenkinsci/plugins/ewm/definitions/Disk.java

```diff
@@ -47,15 +46,15 @@ public String getDiskId() {
 
     @CheckForNull
     public String getName() {
-        return name;
+        return name != null ? name : diskId;
     }
 
     @CheckForNull
     public String getMasterMountPoint() {
         return masterMountPoint;
     }
 
-    @CheckForNull
+    @Nonnull
     public String getPhysicalPathOnDisk() {
         return physicalPathOnDisk;
```

Comment:
I would avoid adding *readResolve()* method at this stage. It's still an alpha version, and the users (if there are any) are aware that things may change and the plugin may break.